### PR TITLE
PHP 7.4 compatibility

### PR DESCRIPTION
**Pull request by ephrin**
- usage of array_key_exists on objects is deprecated used `property_exists` instead.

 - ceased support for PHP < 7

 - code style PSR12

### CODE REVIEW

File: src/getValueFunctions.php [hunk 1]
```diff
@@ -8,36 +8,42 @@
...
  */
-function getValue($doc, $fieldName, $default = null) {
+function getValue($doc, $fieldName, $default = null)
+{
     if ($doc instanceof GetSetInterface) {
         return $doc->getValue($fieldName, $default);
     }
-    if (is_array($doc) || $doc instanceof \ArrayAccess) {
-        return (isset($doc[$fieldName]) || array_key_exists($fieldName, $doc)) ? $doc[$fieldName] : $default;
+    if (is_array($doc)) {
```
Comment (ephrin) [inline src/getValueFunctions.php:40]: The main change is here. Everything else is only CS.

File: src/GetterSetter.php [hunk 2]
```diff
@@ -10,63 +10,74 @@
...
-     * @param string       $fieldName
-     * @param mixed        $value
+     * @param string $fieldName
+     * @param mixed $value
      * @return $this
+     * @throws UnableToSetFieldException
      */
-    public function setValue($data, $fieldName, $value) {
+    public function setValue($data, $fieldName, $value)
+    {
```
Comment (cyrilRevinate) [inline src/GetterSetter.php]: This is not our style of brackets, it should be on the same line as the function.

File: src/GetSet.php [hunk 3]
```diff
@@ -9,11 +9,12 @@
 namespace Revinate\GetterSetter;
 
 
-class GetSet implements GetSetInterface {
+class GetSet implements GetSetInterface
+{
```
Comment (i2aymond) [inline src/GetSet.php:13]: So this is the new CS? class "{" is on the next line, while function "{" is on the same line?

File: src/getValueFunctions.php [hunk 4]
```diff
@@ -51,25 +55,29 @@ function getValue($doc, $fieldName, $default = null) {
...
+        ];
         foreach ($getters as $methodName) {
             if (method_exists($doc, $methodName)) {
                 return $doc->{$methodName}();
             }
         }
 
         // Try the magic methods __get when __isset doesn't exist
-        if (method_exists($doc, '__get') && ! method_exists($doc, '__isset')) {
+        if (method_exists($doc, '__get') && !method_exists($doc, '__isset')) {
```
Comment (i2aymond) [inline src/getValueFunctions.php:72]: We usually keep a space after !, that's the style Revinate keeps. Are we going to change that now?

File: src/setValueFunctions.php [hunk 5]
```diff
@@ -9,41 +9,44 @@
...
+        $doc = [];
     }
-    if (is_array($doc) || $doc instanceof \ArrayAccess) {
+    if (is_array($doc) || $doc instanceof ArrayAccess) {
         $doc[$fieldName] = $value;
         return $doc;
     }
 
-    if (is_object($doc) && ! $doc instanceof \Closure) {
+    if (is_object($doc) && !$doc instanceof Closure) {
```
Comment (i2aymond) [inline src/setValueFunctions.php:49]: Similar comment with !

File: test/AccessTestClass.php [hunk 6]
```diff
@@ -180,7 +181,7 @@ public function setNullValueProtected($nullValueProtected) {
      */
     public function methodGetterSetter() {
         $args = func_get_args();
-        if (! empty($args)) {
+        if (!empty($args)) {
```
Comment (i2aymond) [inline test/AccessTestClass.php:184]: Similar comment with ! and space

File: src/GetSet.php (hunk 3 above)
Comment (ephrin) [reply src/GetSet.php:13]: oh. sorry. left this out of sight

File: src/getValueFunctions.php (hunk 4 above)
Comment (ephrin) [reply src/getValueFunctions.php:72]: i'd suggest to move to PSR12
that was just automatic application of IDE absed PSR12 styles.
they usually are enabled for new projects by default.
I beg your pardon for inconvenience 
we need to decide should we follow these PSR12 or define our own

**Comment by ephrin** [comment]
I'd rather deprecate the library usage as a whole. There are plenty public libraries with wide community.
Looking for replacement candidate.

**Comment by ccarrillot** [comment]
Continue with this merge to check if this works.

**Pull request merged**